src/components/input/input.jsx

```diff
@@ -1,37 +1,65 @@
 import PropTypes from 'prop-types';
 import React from 'react';
+import cautionSvg from '../../svgs/icon-caution-fill.svg';
+import Icon from '../icon/icon.jsx';
 import styles from './input.css';
-import labelPropType from './label-prop-type';
 
-/**
- * Base input component. Accepts any prop that an input can take.
- * @see [RFC](https://github.com/mavenlink/rfc/pull/75)
- * @see [Design spec](https://zpl.io/295Yp7p)
- */
-export default function Input({ 'aria-labelledby': ariaLabelledBy, className, id, ...rest }) {
+function getLabelClassName(className, invalid) {
+  if (className) return className;
+  return invalid ? styles['invalid-label'] : styles.label;
+}
+
+function getClassName(className, invalid) {
+  if (className) return className;
+  return invalid ? styles['invalid-input'] : styles.input;
+}
+
+export default function Input(props) {
   return (
-    <input
-      {...rest}
-      aria-labelledby={ariaLabelledBy}
-      className={className}
-      id={id}
-    />
+    <React.Fragment>
+      <label htmlFor={props.id} className={getLabelClassName(props.cssLabel, props.invalid)}>
+        {props.label}
+      </label>
+      <div className={styles.container}>
+        <input
+          className={getClassName(props.className, props.invalid)}
+          disabled={props.disabled}
+          id={props.id}
+          name={props.name}
+          onChange={props.onChange}
```

Comment:
Do we plan to support something like `onKeyDown`? This allows for things like handling ENTER:

```js
onKeyDown={handleKeyDown}
...
  const handleKeyDown = (e) => {
    if (e.key === 'Enter') {
      e.preventDefault();
      if (inputValue) {
        props.onAdd(inputValue);
        setInputValue('');
      }
    }
  }
```

Are there other events we should allow? Obviously, we should keep surface API lean, but I think the key down use case is compelling enough to consider seriously.

Reply:
I am going to not support it until we have a real use-case. I might be useful in some cases but perhaps that is just another component.